src/unitxt/processors.py

```diff
@@ -277,3 +279,59 @@ def process_value(self, text: Any) -> Any:
 
         except:
             return 0
+
+
+class InferDictsToBinaryLogprobs(FieldOperator):
+    neg_class_name: str
+    pos_class_name: str
+
+    take_logprobs_from_end: bool = False
+    num_logprobs_to_take: int = 3
```

Comment:
Thanks for making the changes.  If I understand right  this attribute is the index of the first token to search for the classes in.  If so the name should reflect it,. Also why start from 3, and not from the first token?

Reply:
This is not the index of the first token, but the total number of tokens to look at. So if the value is 3 we look at index 0,1,2 (in regular case) or -1,-2,-3 (if we look at tokens at the end rather than the beginning). This is because in the beginning/end there are often special characters, newlines etc. Once we get to a token where the class names are candidates we stop and return the values at this index.

Reply:
Ok. Now I understand. Thanks.